Api/Functions/SlackIncoming.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using System.IO;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using System.Net.Http;
+using Merq;
+using NosAyudamos.Events;
+using System.Globalization;
+
+namespace NosAyudamos.Functions
+{
+    class SlackIncoming
+    {
+        const string ApiUrl = "https://slack.com/api/";
+
+        readonly ISerializer serializer;
+        readonly IEnvironment environment;
+        readonly IEventStream events;
+        readonly ILanguageUnderstanding language;
+        readonly HttpClient http;
+        readonly ILogger<SlackIncoming> logger;
+
+        public SlackIncoming(ISerializer serializer, IEnvironment environment, IEventStream events, ILanguageUnderstanding language, HttpClient http, ILogger<SlackIncoming> logger) =>
+            (this.serializer, this.environment, this.events, this.language, this.http, this.logger) = (serializer, environment, events, language, http, logger);
+
+        [FunctionName("slack_interaction")]
+        public async Task<IActionResult> InteractionAsync(
```

Comment:
maybe TrainAsync ?

Reply:
I expect we'll have more interactions, and they would all end up in this same method. Given what we talked, this would end up pushing another event, which would indeed be `Train...`